### PR TITLE
Hotfix for a directory mode and bugs introduced in ignorelines

### PR DESCRIPTION
**Pull request by akuleshov7**

### CODE REVIEW

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt [hunk 1]
```diff
@@ -143,6 +149,11 @@ class WarnPlugin(
         // NOTE: SAVE will pass relative paths of Tests (calculated from testRootConfig dir) into the executed tool
         val fileNamesForExecCmd =
                 warnPluginConfig.wildCardInDirectoryMode?.let {
+                    val directoryPrefix = testConfig
```
Comment (akuleshov7) [inline save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt:152]: Here we also create execCmd incorrectly, will fix tomorrow

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt [hunk 2]
```diff
@@ -143,10 +149,16 @@ class WarnPlugin(
         // NOTE: SAVE will pass relative paths of Tests (calculated from testRootConfig dir) into the executed tool
         val fileNamesForExecCmd =
                 warnPluginConfig.wildCardInDirectoryMode?.let {
-                    // a hack to put only the directory path to the execution command
+                    // a hack to put only the root directory path to the execution command
                     // only in case a directory mode is enabled
-                    "${copyPaths[0].parent!!}$it"
+                    var testRootPath = testConfig.directory.parent ?: testConfig.directory
```
Comment (akuleshov7) [inline save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt:155]: Here we intentionally will skip the case when user provided root directory and particular directory.
In this case execCmd will always run the tool on the root directory. And it is fine and expected

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt [hunk 3]
```diff
@@ -143,10 +149,16 @@ class WarnPlugin(
...
+                    while (testRootPath.parent != null && testRootPath.parent != ".".toPath()) {
```
Comment (akuleshov7) [inline save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt:156]: @kgevorkyan please handle and check "." properly

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt [hunk 4]
```diff
@@ -143,10 +149,16 @@ class WarnPlugin(
...
+                        testRootPath = testRootPath.parent!!
+                    }
+                    "$testRootPath$it"
                 } ?: copyPaths.joinToString(separator = warnPluginConfig.batchSeparator!!)
+
+        logDebug("Constructed file name for execution for warn plugin: $fileNamesForExecCmd")
         val execFlagsAdjusted = resolvePlaceholdersFrom(warnPluginConfig.execFlags, extraFlags, fileNamesForExecCmd)
         val execCmd = "${generalConfig.execCmd} $execFlagsAdjusted"
```
Comment (akuleshov7) [inline save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt:164]: Pls create an issue and assign it on me for running execCmd once

Comment (kgevorkyan) [reply save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt:164]: https://github.com/analysis-dev/save/issues/350

File: save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt (hunk 3 above)
Comment (kgevorkyan) [reply save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt]: fixed

**Review by kgevorkyan — APPROVED** [review]

**Pull request merged**